### PR TITLE
[interp] mixed mode fixes for delegates

### PR DESCRIPTION
**Pull request by lewurm**
Fixes some issues when all DLLs are full-aot'd, but the interpreter is still used for SRE.

Contributes to https://github.com/xamarin/xamarin-macios/issues/5622

### CODE REVIEW

**Comment by luhenry** [comment]
@monojenkins build failed

**Comment by luhenry** [comment]
This breaks `MonoTests.runtime-aot-interp.dynamic-method-churn.exe` all the `FullAOT+Interpreter{+LLVM}` jobs.

- https://jenkins.mono-project.com/job/test-mono-pull-request-arm64-fullaot+interpreter/867/testReport/junit/MonoTests/runtime-aot-interp/dynamic_method_churn_exe/
- https://jenkins.mono-project.com/job/test-mono-pull-request-arm64-fullaot+interpreter+llvm/839/testReport/junit/MonoTests/runtime-aot-interp/dynamic_method_churn_exe/
- https://jenkins.mono-project.com/job/test-mono-pull-request-fullaot+interpreter/940/testReport/junit/MonoTests/runtime-aot-interp/dynamic_method_churn_exe/
- https://jenkins.mono-project.com/job/test-mono-pull-request-fullaot+interpreter+llvm/860/testReport/junit/MonoTests/runtime-aot-interp/dynamic_method_churn_exe/

**Comment by lewurm** [comment]
thanks, already looking into it 🙂

File: mono/mini/interp/interp.c [hunk 1]
```diff
@@ -1454,6 +1451,12 @@ interp_init_delegate (MonoDelegate *del)
 			mono_error_assert_ok (error);
 		}
 	}
+
+	if (!((InterpMethod *) del->interp_method)->transformed && method_is_dynamic (method)) {
```
Comment (BrzVlad) [inline mono/mini/interp/interp.c]: @lewurm What's the reasoning behind transforming the delegate method only if the method is dynamic, during delegate creation ?

Comment (lewurm) [reply mono/mini/interp/interp.c]: Good question. I observed two requirements:

1. If the (eventual) delegate body is part of the CIL image, the compilation error should be materialize as managed exception upon the first invocation. See `mono/tests/delegate15.cs`: https://github.com/mono/mono/blob/df37bb7a7c874fa33e1aa5ddf3648df4aec5d51e/mono/tests/delegate15.cs
2. If the delegate body is generated _at runtime_, such compilation errors are expected at delegate creation. See `MemberBindTests.cs`: https://github.com/mono/corefx/blob/68f67ad6ced8b74a65a1a5617a4c761bb203a25b/src/System.Linq.Expressions/tests/MemberInit/MemberBindTests.cs#L165-L178

This was previously covered by `create_delegate_method_ptr ()`: https://github.com/mono/mono/blob/85c8e85476fcc2076e3b6e362c77c62f463340c5/mono/mini/mini-runtime.c#L3555-L3570

**Pull request merged**